code/datums/components/_component.dm

```diff
@@ -237,7 +237,10 @@
 
 	if(dm != COMPONENT_DUPE_ALLOWED)
 		if(!dt)
-			old_comp = GetExactComponent(nt)
+			if(ispath(nt))
+				old_comp = GetExactComponent(nt)
+			else
+				old_comp = GetExactComponent(nt.type)

```

Comment:
Prety sure we shouldnt be doing addcomponent with instances anymore. You want `target.TakeComponent(comp_instance)` instead

Reply:
@Garen7

Reply:
Sorry I was busy I’m going to fix it tomorrow

Reply:
I double checked this and I was mistaken earlier. However do `nt = new_comp.type` in the else of the ispath check above instead.